transcom/mymove
# B 19941 int

## [B-19941](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19941)

## Summary

This story makes it such that an office user can no longer request a diversion in the application prior to the actual pickup date for a shipment. Additionally, it displays the diversion reason in the shipment information on the MTO office page.

Is there anything you would like reviewers to give additional scrutiny?
Please ensure all client tests are passing. Additionally, the utils changed are the result of merge conflicts. If this change should be reverted, please let me know and I'll take care of it.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the application as a TOO and find an HHG Shipment to use to test, or get one ready to test by logging in as a Service Counselor and completing counseling first. I have been using move code BILWEI.
2. With the move code you've just found, log in as a Prime Sim user. 
3. Go to that move code as a prime sim, edit the shipment, and change the actual pickup date to a day in the future.
4. Log back in as a TOO, and go to view the same move. On the Move Task Order tab, scroll down to the addresses card and click "Request Diversion".
5. Verify that even after entering a diversion reason, the button to request a diversion is inactive and does not submit. Note that there's no message explaining why it will not submit, that is per PO recommendation. This is AC 1. 
6. Log back in as a Prime Sim user and change back the actual pickup date to some time in the past, or the same day as testing. 
7. Log back in as a TOO, and request a diversion with a reason, and note that it will be submittable this time. Submit the diversion.
8. As a TOO on this MTO page, verify that a diversion has been requested and that the diversion reason is located in the important shipment dates card.
9. Log back in as a Prime Sim and accept the diversion.
10. Log back in as a TOO and notice that on the MTO page, the diversion requested reason has turned into the diversion approved reason. This is AC 2.

### Frontend


## Screenshots
<img width="1266" alt="image" src="https://github.com/transcom/mymove/assets/148906585/a34a94da-38ae-4134-87fb-4025e736737b">
<img width="1266" alt="image" src="https://github.com/transcom/mymove/assets/148906585/6cc5fcaa-ec71-4385-ae11-32b7ab722467">
<img width="678" alt="image" src="https://github.com/transcom/mymove/assets/148906585/e3e30d1f-8911-45f4-9e3d-1aaa1f2bea19">
